src/if_statements.py

```diff
@@ -786,6 +1039,10 @@ def build_flowgraph_between(
                 curr_start = imm_pdom
                 continue
 
+        if curr_start in context.emitted_nodes and is_empty_goto(curr_start, end):
```

Comment:
Yeah I should explain this more. It shouldn't result in any missing `emit_gotos` because `curr_start` is already emitted.

These irregular switches end up with a lot of empty `BasicNode`s which just act as gotos to `end`. If we've hit one, we don't need to emit any more goto/return statements: the return value of `build_flowgraph_between` will be terminated with a `}` or `break;`. 

Example function w/ this check removed, where the end of each case block is a `goto block_41;` and all that block does is `break;`:
```diff
[INFO] Output of ../oot/asm/non_matchings/overlays/actors/ovl_Boss_Ganon/func_808E2544.s changed! Diff:
--- 
+++ 
@@ -152,13 +152,14 @@

             arg0->unk_1C2 = 1;
             arg0->speedXZ = 0.0f;
         }
+block_41:
         break;
     case 1:
         Math_ApproachZeroF(arg0 + 0x1CC, 1.0f, 40.0f);
         if (arg0->unk_1CC == 0.0f) {
             Actor_Kill(arg0);
         }
-        break;
+        goto block_41;
     case 10:
         arg0->unk_1C2 = 0xB;
         arg0->unk_1B6 = 0xE;
@@ -252,7 +253,7 @@

                 }
             }
         }
-        break;
+        goto block_41;
     case 12:
         arg0->speedXZ = 20.0f;
         temp_f12_5 = arg0->unk_1F0 - arg0->world.pos.x;
@@ -283,7 +284,7 @@

             arg0->unk_1C2 = 1;
             arg0->speedXZ = 0.0f;
         }
-        break;
+        goto block_41;
     }
     if ((s32) arg0->unk_1C2 >= 0xB) {
         temp_f12_7 = arg0->world.pos.x;
```

Example function w/ this check removed that doesn't have a switch statement, but still benefits from this change
```diff
[INFO] Output of ../oot/asm/non_matchings/code/ucode_disas/UCodeDisas_Disassemble.s changed! Diff:
--- 
+++ 
@@ -1563,9 +1563,11 @@

                 if (sp370.unk_2 == 0) {
                     if (temp_t0_12 != 0) {
                         osSyncPrintf("count_gsDPNoOpQuiet(),", temp_a1_3);
+                        goto block_253;
                     }
                 } else if (temp_t0_12 != 0) {
                     osSyncPrintf("count_gsDPNoOpVerbose(),", temp_a1_3);
+block_253:
                 }
                 arg0->enableLog = (s32) sp370.unk_2;
                 phi_t0_5 = (s32) sp370.unk_2;
```